app/src/main/java/com/hedvig/app/feature/marketpicker/LanguageAdapter.kt

```diff
@@ -0,0 +1,123 @@
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = ViewHolder(
+        LayoutInflater.from(parent.context)
+            .inflate(R.layout.language_item_new, parent, false)
+    )
+
+    override fun getItemCount() = 2
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        languageViewModel.selectedLanguage.postValue(null)
+        when (position) {
+            LOCAL -> {
+                when (selectedMarket) {
+                    Market.SE -> {
+                        holder.apply {
+                            language.text =
+                                language.resources.getString(R.string.SETTINGS_LANGUAGE_SWEDISH)
+                            parent.setHapticClickListener {
+                                languageViewModel.selectLanguage(Language.SV_SE)
+                            }
+                        }
+                    }
+                    Market.NO -> {
+                        holder.apply {
+                            language.text = language.resources.getString(R.string.norwegian)
+                            parent.setHapticClickListener {
+                                languageViewModel.selectLanguage(Language.NB_NO)
+
+                            }
+                        }
+                    }
+                }
+            }
+            EN -> holder.apply {
+                language.text =
+                    language.resources.getString(R.string.SETTINGS_LANGUAGE_ENGLISH)
+            }
+        }
+
+        if (position == 0 && holder.button.isChecked) {
+            lastChecked = holder.button
+            lastCheckedPos = 0
+        }
+
+        holder.parent.setHapticClickListener { v ->
+            when (position) {
+                EN -> languageViewModel.selectLanguage(Language.EN_SE)
+                LOCAL -> {
+                    when (selectedMarket) {
+                        Market.SE -> languageViewModel.selectLanguage(Language.SV_SE)
+                        Market.NO -> languageViewModel.selectLanguage(Language.NB_NO)
+                    }
+                }
+            }
+            val rb = v.radioButton as RadioButton
+            rb.isChecked = true
+            rb.background = rb.context.getDrawable(R.drawable.ic_radiob_button_checked)
+            animateRadioButton(holder)
+            if (rb.isChecked) {
+                if (lastChecked != null) {
+                    if (lastCheckedPos != position) {
+                        lastChecked!!.background =
+                            rb.context.getDrawable(R.drawable.ic_radio_button_unchecked)
+                        lastChecked!!.isChecked = false
+                        animateRadioButton(holder)
+                    }
+                }
+                lastChecked = rb
+                lastCheckedPos = position
+            } else lastChecked = null
+        }
+    }
+
+    private fun animateRadioButton(holder: ViewHolder) {
+        val button = holder.button
```

Comment:
```suggestion
        holder.button.apply {
```

No local variable needed, and we can remove the `button`-prefix on every line 🙂